docs/architecture-and-design.md

```diff
@@ -387,10 +392,60 @@ Most [Intents](#intents) are easy to break down into  [Primitives](#primitives),
+The [Visualizer](#visualizer) is exactly what it sounds like: A visualizion of our [AI](#ai). It provides a GUI that shows us the state of the [World](#world), and is also able to display extra information that the [AI](#ai) would like to show. For example, it can show the planned paths of each friendly robot on the field, or highlight which enemy robots it thinks are a threat. Furthermore, it displays any warnings or status messages from the robots, such as if a robot is low on battery.
 
-The `Visualizer` also lets us control the [AI](#ai) by setting [Dynamic Parameters](#dynamic-parameters). Through the `Visualizer`, we can manually choose what strategy the [AI](#ai) should use, what colour we are playing as (yellow or blue), and tune more granular behaviour such as how close an enemy must be to the ball before we consider them a threat.
+The [Visualizer](#visualizer) also lets us control the [AI](#ai) by setting [Dynamic Parameters](#dynamic-parameters). Through the [Visualizer](#visualizer), we can manually choose what strategy the [AI](#ai) should use, what colour we are playing as (yellow or blue), and tune more granular behaviour such as how close an enemy must be to the ball before we consider them a threat.
+
+The [Visualizer](#visualizer) is connected to the rest of the system using the [Observer Design Pattern](#observer-design-pattern). It observes Subjects that contain information it wants to display, such as the [World(#world) or [DrawFunctions](#draw-functions).
+
+The [Visualizer](#visualizer) is implemented using [Qt](https://www.qt.io/), a C++ library for creating cross-platform GUIs. The general documentation for [Qt](https://www.qt.io/) can be found [here](https://doc.qt.io/qt-5/index.html). The most important fundamentals are:
+* [Signals and Slots](https://doc.qt.io/qt-5/signalsandslots.html)
+* [QtCreator](https://doc.qt.io/qtcreator/creator-using-qt-designer.html) (specifically for Widget-based applications)
+* [Widgets](https://doc.qt.io/qt-5/qtwidgets-index.html)
+
+The [Visualizer](#visualizer) is made up of 2 major components: The [QApplication](https://doc.qt.io/qt-5/qapplication.html) and `Visualizer Widget` that contains all of the graphical components used in the [Visualizer](#visualizer), and the `VisualizerWrapper`. The `VisualizerWrapper` runs the [QApplication](https://doc.qt.io/qt-5/qapplication.html) in a separate thread, so that Qt can run its event loop and handle events and rendering without blocking our main thread.
```

Comment:
Assuming I understand this correct, the `VisualizerWrapper` contains an instance of the `VisualizerWidget`. If that's the case, describing these as "two major components" is a bit confusing to me, in that it made me think of them as being separate.

Reply:
See above reply.